op-chain-ops/genesis/db_migration.go

```diff
@@ -26,28 +92,55 @@ func MigrateDB(db vm.StateDB, config *DeployConfig, l1Block *types.Block, l2Addr
 		return err
 	}
 
-	// Convert all of the messages into legacy withdrawals
-	messages := make([]*crossdomain.LegacyWithdrawal, 0)
-	for _, msg := range migrationData.OvmMessages {
-		wd, err := msg.ToLegacyWithdrawal()
-		if err != nil {
-			return err
-		}
-		messages = append(messages, wd)
+	err = crossdomain.MigrateWithdrawals(withdrawals, db, &l2Addrs.L1CrossDomainMessengerProxy, &l2Addrs.L1StandardBridgeProxy)
+	if err != nil {
+		return err
 	}
-	for _, msg := range migrationData.EvmMessages {
-		wd, err := msg.ToLegacyWithdrawal()
-		if err != nil {
-			return err
-		}
-		messages = append(messages, wd)
+
+	addrs := migrationData.Addresses()
+	if err := ether.MigrateLegacyETH(ldb, addrs, migrationData.OvmAllowances, 0); err != nil {
+		return err
 	}
 
-	if err := crossdomain.MigrateWithdrawals(messages, db, &l2Addrs.L1CrossDomainMessengerProxy, &l2Addrs.L1StandardBridgeProxy); err != nil {
+	if !commit {
+		return nil
+	}
+
+	root, err := db.Commit(true)
+	if err != nil {
 		return err
 	}
 
-	// TODO: use migration data to double check things
+	// Create the bedrock transition block
+	bedrockHeader := &types.Header{
+		ParentHash:  header.Hash(),
+		UncleHash:   types.EmptyUncleHash,
+		Coinbase:    config.L2GenesisBlockCoinbase,
+		Root:        root,
+		TxHash:      types.EmptyRootHash,
+		ReceiptHash: types.EmptyRootHash,
+		Bloom:       types.Bloom{},
+		Difficulty:  (*big.Int)(config.L2GenesisBlockDifficulty),
+		Number:      new(big.Int).Add(header.Number, common.Big1),
+		GasLimit:    (uint64)(config.L2GenesisBlockGasLimit),
+		GasUsed:     (uint64)(config.L2GenesisBlockGasUsed),
+		Time:        uint64(config.L2OutputOracleStartingTimestamp),
+		Extra:       config.L2GenesisBlockExtraData,
+		MixDigest:   config.L2GenesisBlockMixHash,
+		//Nonce:       (types.BlockNonce)(config.L1GenesisBlockNonce),
```

Comment:
TODO: wasn't sure how to inline this, going to need to binary.WriteUint64 i think